pkg/asset/installconfig/aws/validation.go

```diff
@@ -56,6 +59,17 @@ func validatePlatform(ctx context.Context, meta *Metadata, fldPath *field.Path,
 	return allErrs
 }
 
+func validatePublicZoneID(ctx context.Context, fldPath *field.Path, platform *awstypes.Platform, publish types.PublishingStrategy) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if publish == types.InternalPublishingStrategy {
+		return allErrs
+	}
```

Comment:
i think this should be an error, public zone id being set with internal is incorrect configuration right?

also again 
`pkg/asset/installconfig/aws/validations.go` -> validations that need AWS API acesss.
`pkg/types/aws/validations/platform.go` -> structural validations that define valid values. 
    making sure public zone id is not set with internal and that public zone id needs to be atleast 2 characters long belong in structural bucket.

Reply:
Fixed!